src/common/k8s-api/endpoints/nodes.api.ts

```diff
@@ -212,13 +212,9 @@ export class Node extends KubeObject {
   }
 
   getOperatingSystem(): string {
-    const label = this.getLabels().find(label => label.startsWith("kubernetes.io/os="));
-
-    if (label) {
-      return label.split("=", 2)[1];
-    }
-
-    return "linux";
+    return this.metadata?.labels?.["kubernetes.io/os"]
```

Comment:
There is no need to serialize all the labels and then split them again, I also added the status fallback since that is documented to be present as well.

Reply:
previously also `beta.kubernetes.io/os` label has been taken into an account.